src/ucs/sys/sys.c

```diff
@@ -488,38 +487,77 @@ void ucs_get_mem_page_size(void *address, size_t size, size_t *min_page_size_p,
             continue;
         }
 
-        if (start > (uintptr_t)address + size) {
-            /* the scanned range is after memory range of interest - stop */
-            break;
-        }
-        if (end <= (uintptr_t)address) {
-            /* the scanned range is still before the memory range of interest */
-            continue;
+        if (address != NULL) {
+            if (start > (uintptr_t)address + size) {
+                /* the scanned range is after memory range of interest - stop */
+                break;
+            }
+            if (end <= (uintptr_t)address) {
+                /* the scanned range is still before the memory range of interest */
+                continue;
+            }
         }
 
+        info.start = start;
+        info.end   = end;
```

Comment:
need to reset all "info" fields here, and not in line 472, to avoid leftovers across regions